miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -568,20 +586,26 @@ export.get_procedure_info
     # => [index]
 
     # get procedure pointer
-    mul.2 exec.memory::get_acct_procedures_section_ptr add dup add.1
-    # => [offset_ptr, proc_ptr]
+    mul.2 exec.memory::get_acct_procedures_section_ptr add dup add.1 swap
+    # => [metadata_ptr, proc_ptr]
 
     # load procedure information from memory
-    mem_load swap padw movup.4 mem_loadw
-    # => [PROC_ROOT, storage_offset]
+    padw movup.4 mem_loadw padw movup.8 mem_loadw
+    # => [METADATA, PROC_ROOT]
+
+    # keep only relevant data
+    swap drop swap drop swap movdn.5 movdn.5
+    # => [PROC_ROOT, storage_offset, storage_size]
```

Comment:
If `METADATA` was `[storage_offset, storage_size, 0, 0]`, this could could be just:
```
drop drop movdn.5 movdn.5
```